src/autocollapse/autocollapse.js

```diff
@@ -25,8 +27,9 @@ function collapseIfNeeded(section) {
 
 async function collapseIfNeededAsync(section) {
```

Comment:
1. Since we now have another condition to evaluate when deciding whether we need to collapse the file or not, I think it would be more readable if we invert the `if` statement now,  and add the condition to a variable.
1. Also, I'm trying to use as much native API as possible, unless absolutely necessary to use a library (jQuery in this case).

... so in my proposal it would look like this:

```js
async function collapseIfNeededAsync(section) {
    const filename = section.getAttribute('data-filename').trim();
    const isDeleted = section.querySelector('h1.filename span.diff-entry-lozenge.aui-lozenge-error');

    const shouldCollapse = ig.ignores(filename) || (autocollapseDeletedFiles && isDeleted);

    if (shouldCollapse) {
        await elementReady('.__refined_bitbucket_collapse_diff_button', {target: section});
        toggleDiff(section);
    }
}
```